src/http/client/response.cr

```diff
@@ -131,14 +140,15 @@ class HTTP::Client::Response
       raise "Invalid HTTP status code: #{pieces[1]}"
     end
 
+    status = HTTP::Status.from_code(status_code)
```

Comment:
This is wrong. We shouldn't raise if we receive an invalid http status code in the response. Please use new here.

Reply:
That depends on what is considered invalid, see thread above. It should definitely raise if the status code is not a three-digit number.

We might consider allowing to receive `600..999` and maybe `000`..`099` yet let these ranges fail for `from_code`. But I think that's to complicated having different restrictions.

It's probably best if `000..999` is allowed everywhere.

Reply:
Oh, but values < 100 need extra care when printing. Would be printf `%03d` for inserting leading zeros to make it three digits always.